### PR TITLE
feat(ui) Display consistent search results in embedded searches

### PR DESCRIPTION
**Pull request by chriscollins3456**
Makes an update so that embedded searches display the same search result card as regular search results. That means we show all the same info (usage stats, owners, all that good stuff).

Previously `EntityNameList` was being used in two different contexts - (1) embedded list searches (2) home page recommendation modules.
The tough thing is that in order to display the regular search card, we need the whole `SearchResult` object which we have from embedded list searches but not in home page recommendations. So what I did was pull out the contents of `EntityNameList` and create a new component that is essentially the same but instead takes in `SearchResult` objects instead of `Entity` objects and use the entity registry's `renderSearch` method which displays the full search result card.

Then I stripped `EntityNameList` of any unused functionality so it only does what is needed for it from the recommendation modules. Therefore the diff is a little misleading here.

Screenshot:
image

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by joshuaeilers — APPROVED** [review]
Looks reasonable - tested out the different variations where it's rendered already yeah? Like the Recommendations + results.

**Comment by chriscollins3456** [comment]
> Looks reasonable - tested out the different variations where it's rendered already yeah? Like the Recommendations + results.

indeed i have

File: datahub-web-react/src/app/entity/shared/components/styled/search/EntitySearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                            {isSelectMode && (
+                                <StyledCheckbox
+                                    checked={selectedEntityUrns.indexOf(entity.urn) >= 0}
+                                    onChange={(e) =>
+                                        onSelectEntity({ urn: entity.urn, type: entity.type }, e.target.checked)
+                                    }
+                                />
+                            )}
+                            {entityRegistry.renderSearchResult(entity.type, searchResult)}
+                            {entityAction && <EntityAction urn={entity.urn} type={entity.type} />}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/components/styled/search/EntitySearchResults.tsx:90]: we can test this using the metadata tests flow on acryl

File: datahub-web-react/src/app/recommendations/renderer/component/EntityNameList.tsx [hunk 2]
```diff
@@ -54,78 +49,23 @@ const ListItem = styled.div<{ isSelectMode: boolean }>`
...
-    additionalPropertiesList,
-    entities,
-    onClick,
-    isSelectMode,
-    selectedEntities = [],
-    setSelectedEntities,
-    bordered = true,
-    entityAction,
-}: Props) => {
+export const EntityNameList = ({ entities, onClick }: Props) => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/recommendations/renderer/component/EntityNameList.tsx:62]: Are we still using this component anywhere?

**Pull request merged**